File: app/src/main/java/io/github/qwqgong/androidcyaml/TaskVisibilityController.java
package io.github.qwqgong.androidcyaml;

import android.app.Activity;
import android.app.ActivityManager;

final class TaskVisibilityController {
    private final Activity activity;

    TaskVisibilityController(Activity activity) {
        this.activity = activity;
    }

    void setHiddenFromRecents(boolean hidden) {
        ActivityManager manager = activity.getSystemService(ActivityManager.class);
        if (manager == null) {
            return;
        }
        for (ActivityManager.AppTask task : manager.getAppTasks()) {
            if (task.getTaskInfo().taskId == activity.getTaskId()) {
                task.setExcludeFromRecents(hidden);
                return;
            }
        }
    }
}

File: app/src/main/java/io/github/qwqgong/androidcyaml/TaskVisibilityController.kt
package io.github.qwqgong.androidcyaml

import android.app.Activity
import android.app.ActivityManager

class TaskVisibilityController(private val activity: Activity) {
    fun setHiddenFromRecents(hidden: Boolean) {
        val manager = activity.getSystemService(ActivityManager::class.java) ?: return
        for (task in manager.appTasks) {
            if (task.taskInfo?.taskId == activity.taskId) {
                task.setExcludeFromRecents(hidden)
                return
            }
        }
    }
}
